experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,662 @@
+        return G
+    end
+end
+
+function Base.show(io::IO, G::ReductiveGroup)
+    io = pretty(io)
+    if G.group[1] == :SL
+        print(io, "Reductive group ", G.group[1], G.group[2])
+    else
+        println(io, "Torus of rank ", group(G)[2])
+        print(IOContext(io, :supercompact => true), Indent(), "over the field ", field(G))
+        print(io, Dedent())
+    end
+end
+
+reductive_group(sym::Symbol, m::Int, R::MPolyRing) = ReductiveGroup(sym,m,R)
+reductive_group(sym::Symbol, m::Int, F::Field) =  ReductiveGroup(sym,m,F)
+group(G::ReductiveGroup) = G.group
+field(G::ReductiveGroup) = G.field
+reynolds_operator(G::ReductiveGroup) = G.reynolds_operator
+group_ideal(G::ReductiveGroup) = G.group_ideal
+canonical_representation(G::ReductiveGroup) = G.canonical_representation
+natural_representation(G::ReductiveGroup) = G.canonical_representation
+
+#####################
+#Representation of Reductive Groups. Embeds SLm in GLn. 
+#####################
+
+mutable struct RepresentationReductiveGroup
+    group::ReductiveGroup
+    rep_mat::AbstractAlgebra.Generic.MatSpaceElem
+    sym_deg::Tuple{Bool, Int}
+    reynolds_v::Function
+
+    weights::Vector{Vector{ZZRingElem}}
+    
+    #representation of group G over symmetric degree d
+    function RepresentationReductiveGroup(G::ReductiveGroup, d::Int)
+        R = new()
+        R.group = G
+        R.rep_mat = rep_mat_(G, d)
+        R.sym_deg = (true, d)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+
+    #function RepresentationReductiveGroup(G::ReductiveGroup, W::Union{ZZMatrix, Matrix{<:Integer}, Vector{<:Integer}})
+    function RepresentationReductiveGroup(G::ReductiveGroup, W::Vector{Vector{ZZRingElem}})
+        R = new()
+        R.group = G
+        R.weights = W
+        return R
+    end
+    
+    #matrix M is the representation matrix. does not check M.
+    function RepresentationReductiveGroup(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem)
+        @req base_ring(M) == base_ring(G.group_ideal) "Group ideal and representation matrix must have same parent ring"
+        R = new()
+        R.group = G
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+#representation_reductive_group(G::ReductiveGroup, d::Int) = RepresentationReductiveGroup(G, d)
+representation_reductive_group(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem) = RepresentationReductiveGroup(G,M)
+group(R::RepresentationReductiveGroup) = R.group
+
+function representation_on_forms(G::ReductiveGroup, d::Int)
+    @assert G.group[1] == :SL
+        return RepresentationReductiveGroup(G, d)
+end
+
+function representation_from_weights(G::ReductiveGroup, W::Union{ZZMatrix, Matrix{<:Integer}, Vector{<:Int}})
+    @assert G.group[1] == :torus
+    V = Vector{Vector{ZZRingElem}}()
+    if typeof(W) <: Vector
+        G.group[2] == 1 || @error("Incompatible weights")
```

Comment:
Don't use `@error` it just prints a text. I think you meant `error`:
```suggestion
        G.group[2] == 1 || error("Incompatible weights")
```
You could also rewrite this as
```suggestion
        G.group[2] == ncols(W) || error("Incompatible weights")
```
and move it befire the `if/else`